### PR TITLE
KEYCLOAK-7413 Fuse 7 documentation

### PR DESCRIPTION
**Pull request by hmlnarik**

### CODE REVIEW

File: securing_apps/topics/oidc/java/fuse7-adapter.adoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{project_name} supports securing your web applications running inside https://developers.redhat.com/products/fuse/overview/[JBoss Fuse 7].
+
+JBoss Fuse 7 leverages Undertow adapter which is essentially the same as 
+ifeval::[{project_community}==true]
+<<_jboss_adapter,EAP 7 / WildFly 10 Adapter>>
+endif::[]
+ifeval::[{project_product}==true]
+<<_jboss_adapter,JBoss EAP 7 Adapter>>
+endif::[]
+as {fuse7Version} is bundled with http://undertow.io/[Undertow HTTP engine] under the covers and Undertow is used for running
```
Comment (vramik) [inline securing_apps/topics/oidc/java/fuse7-adapter.adoc]: unnecessary new line character

**Review by mposolda — DISMISSED** [review]

File: securing_apps/topics/oidc/java/fuse7-adapter.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+various kinds of web applications.
+
+WARNING: The only supported versions of Fuse 7 is {fuse7Version}. If you use earlier versions of Fuse 7, it is possible that some functions will not work correctly. In particular, integration will not work at all for earlier versions of Fuse 7 than 7.0.1.
+
+Security for the following items is supported for Fuse:
+
+* Classic WAR applications deployed on Fuse with Pax Web War Extender
+* Servlets deployed on Fuse as OSGI services with Pax Web Whiteboard Extender
```
Comment (hmlnarik) [inline securing_apps/topics/oidc/java/fuse7-adapter.adoc]: From @grgrzybek:
> and additionally servlets registered through org.osgi.service.http.HttpService#registerServlet() which is standard, OSGi Enterprise HTTP Service

**Review by matthewhelmke — DISMISSED** [review]

**Comment by hmlnarik** [comment]
Addressed the two minor comments from above.

**Comment by hmlnarik** [comment]
The build failure is caused by server refusing connection for link (https://self-issued.info/docs/draft-ietf-jose-json-web-key.html). The error is unrelated to this PR.

**Review by pdrozd — APPROVED** [review]

**Comment by hmlnarik** [comment]
Thanks for all the reviews!

File: securing_apps/topics/oidc/java/fuse-adapter.adoc [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 [[_fuse_adapter]]
-==== JBoss Fuse Adapter
+==== JBoss Fuse 6 Adapter
 
-Currently {project_name} supports securing your web applications running inside https://developers.redhat.com/products/fuse/overview/[JBoss Fuse].
+{project_name} supports securing your web applications running inside https://developers.redhat.com/products/fuse/overview/[JBoss Fuse 6].
 
 ifeval::[{project_community}==true]
-It leverages <<_jetty9_adapter,Jetty 9 adapter>> as {fuseVersion} is bundled with http://www.eclipse.org/jetty/[Jetty 9.2 server]
+JBoss Fuse 6 leverages <<_jetty9_adapter,Jetty 9 adapter>> as {fuseVersion} is bundled with http://www.eclipse.org/jetty/[Jetty 9.2 server]
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse-adapter.adoc:8]: Only Fuse 6.3. Fuse 6.2.1 still uses Jetty 8.2

File: securing_apps/topics/oidc/java/fuse-adapter.adoc [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 under the covers and Jetty is used for running various kinds of web applications.
 endif::[]
 
-WARNING: The only supported version of Fuse is {fuseVersion}. If you use earlier versions of Fuse, it is possible that some functions will not work correctly. In particular, the http://hawt.io[Hawtio] integration will not work with earlier versions of Fuse.
+WARNING: The only supported version of Fuse 6 is {fuseVersion}. If you use earlier versions of Fuse 6, it is possible that some functions will not work correctly. In particular, the http://hawt.io[Hawtio] integration will not work with earlier versions of Fuse 6.
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse-adapter.adoc:12]: But if you mention about minimal Fuse version here (assuming 6.3.Rx), I'm fine

File: securing_apps/topics/oidc/java/fuse7-adapter.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Security for the following items is supported for Fuse:
+
+* Classic WAR applications deployed on Fuse with Pax Web War Extender
+* Servlets deployed on Fuse as OSGI services with Pax Web Whiteboard Extender and additionally servlets registered through
+  org.osgi.service.http.HttpService#registerServlet() which is standard OSGi Enterprise HTTP Service
+* http://camel.apache.org/[Apache Camel] Undertow endpoints running with the http://camel.apache.org/undertow.html[Camel Undertow] component
+* http://cxf.apache.org/[Apache CXF] endpoints running on their own separate Undertow engine
+* http://cxf.apache.org/[Apache CXF] endpoints running on the default engine provided by the CXF servlet
+* SSH and JMX admin access
+* http://hawt.io[Hawtio administration console]
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse7-adapter.adoc:27]: Exactly ;)

File: securing_apps/topics/oidc/java/fuse7-adapter.adoc [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+===== Securing Your Web Applications Inside Fuse 7
+
+You must first install the {project_name} Karaf feature. Next you will need to perform the steps according to the type of application you want to secure.
+All referenced web applications require injecting the {project_name} Undertow authentication mechanism into the underlying web server. The steps to achieve this depend on the application type. The details are described below.
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse7-adapter.adoc:32]: Technically (pax-web) the mechanism is injected per web application/war/web context/`io.undertow.servlet.api.DeploymentInfo`/`javax.servlet.ServletContext`/`org.ops4j.pax.web.service.undertow.internal.Context` not per entire web server. There may be contexts without keycloak authentication mechanism being injected.

File: securing_apps/topics/oidc/java/fuse7/fuse-admin.adoc [hunk 7]
```diff
@@ -0,0 +1,83 @@
+
+[[_fuse7_adapter_admin]]
+===== Securing Fuse Administration Services
+
+====== Using SSH Authentication to Fuse Terminal
+
+{project_name} mainly addresses use cases for authentication of web applications; however, if your other web services and applications are protected
+with {project_name}, protecting non-web administration services such as SSH with {project_name} credentials is a best pracrice. You can do this using the JAAS login module, which allows remote connection to {project_name} and verifies credentials based on
```
Comment (grgrzybek) [inline securing_apps/topics/oidc/java/fuse7/fuse-admin.adoc:8]: s/pracrice/practice/

**Review by grgrzybek — COMMENTED** [review]
OK @hmlnarik, I added my comments to the review.

**Pull request merged**